source/manual/govuk-in-aws.html.md

```diff
@@ -0,0 +1,89 @@
+
+### Hostnames
+
+Traditionally we hardcoded hostnames and IPs on each instance in `/etc/hosts`. In AWS, we are making use of [Auto Scaling Groups (ASG)](http://docs.aws.amazon.com/autoscaling/latest/userguide/AutoScalingGroup.html) and [Elastic Load Balancing](https://aws.amazon.com/elasticloadbalancing/) to connect to instances.
+
+Traditionally you would see hostnames similar to:
+
+```
+backend-1.backend
+frontend-1.frontend
+puppetmaster-1.management
+```
+
+Hostnames are now automatically generated by DHCP, and refer to the IP address and [region](http://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/Concepts.RegionsAndAvailabilityZones.html) that the instance belongs to:
+
+```
+ip-10-1-4-100.eu-west-1.compute.internal
+```
+
+Please see [the documentation]() about accessing the environment.
+
+### Databases
+
+#### PostgreSQL and MySQL
+
+We are using [Amazon Relational Database Service (RDS)](https://aws.amazon.com/rds/) to host PostgreSQL and MySQL databases.
+
+To run Puppet against these databases, we have a new instance class: [db_admin](https://github.com/alphagov/govuk-puppet/blob/master/modules/govuk/manifests/node/s_db_admin.pp)
+
+Both PostgreSQL and MySQL databases are managed through this instance. They are also used to take textual backups, which are then stored in an [Amazon S3](https://aws.amazon.com/s3/) bucket.
+
+Transition has it's own class for management: [transition_db_admin](https://github.com/alphagov/govuk-puppet/blob/master/modules/govuk/manifests/node/s_transition_db_admin.pp)
+
+Please see [the documentation]() about administering RDS databases.
+
+#### Redis
+
+We are using [Amazon Elasticache](https://aws.amazon.com/elasticache/) instead of managing our own Redis
+instances.
+
+### Architecture changes
+
+#### Removal of LB tiers
+
+Due to the use of [Elastic Load Balancing](https://aws.amazon.com/elasticloadbalancing/) we no longer have a need to maintain
+our own NGINX load balancers, and so these have been removed from the stack. See the [related ADR](https://github.com/alphagov/govuk-aws/blob/master/doc/architecture/decisions/0026-remove-load-balancer-tier.md) for further details.
+
+#### Merging MySQL databases
+
+Traditionally we had a separate MySQL server for Whitehall. Rather than manage multiple RDS instances,
+we have merged this into the main MySQL server. See the [relevant ADR](https://github.com/alphagov/govuk-aws/blob/master/doc/architecture/decisions/0019-centralise-mysql-databases.md) for details.
+
+### Deploying infrastructure
+
+[Terraform](https://terraform.io) is used to manage our [Infrastructure as Code](https://en.wikipedia.org/wiki/Infrastructure_as_Code). This code is stored
+in [govuk-aws](https://github.com/alphagov/govuk-aws).
+
+See the [documentation]() to make and deploy changes to the infrastructure.
+
+### Automated application deployments
+
+If an EC2 instance is terminated, it will be automatically rebuilt by the ASG. If an instance runs deployable applications,
```

Comment:
What is ASG?

Reply:
Auto Scaling Group. It was referenced previously in the document, but I will specify.